search/query/regexp.go

```diff
@@ -76,13 +78,13 @@ func (q *RegexpQuery) Validate() error {
 
 func (q *RegexpQuery) compile() error {
 	if q.compiled == nil {
-		// require that pattern be anchored to start and end of term
+		// require that pattern NOT be anchored to start and end of term
 		actualRegexp := q.Regexp
-		if !strings.HasPrefix(actualRegexp, "^") {
-			actualRegexp = "^" + actualRegexp
+		if strings.HasPrefix(actualRegexp, "^") {
+			actualRegexp = actualRegexp[1:] // remove leading ^
 		}
-		if !strings.HasSuffix(actualRegexp, "$") {
-			actualRegexp = actualRegexp + "$"
+		if strings.HasSuffix(actualRegexp, "$") {
+			actualRegexp = actualRegexp[:len(actualRegexp)-1] // remove trailing $
 		}
```

Comment:
Actually I think its a reasonable concern.  In particular I remember one index encoding which put $ by itself at the end of terms.  Totally unrelated to this case, but still makes me nervous.

I think instead I'll remove the code that trims off $ from the end.  In the cases we know about the presence of $ does not interfere with LiteralPrefix().  I had removed it to handle it consistently with ^, but issue of possible escaping makes it trickier for sure.